manifests/router/service.pp

```diff
@@ -11,12 +11,18 @@
     hasrestart => true,
   }
 
-  if $::qpid::router::open_file_limit and $::systemd {
+  if $::qpid::router::open_file_limit != 1 {
```

Comment:
You can leave out the `!= 1` and just test for undef.

Reply:
I cannot because if you set this to undef, the systemd:system_limits bombs out if trying to ensure absent. I would say this is a bug in the systemd:system_limit logic and this my work around.

Reply:
Ah, nasty.

Reply:
Maybe it's better to specify two resources, one with limits and absent and the other without?